internal/statestore/redis.go

```diff
@@ -538,6 +538,13 @@ func (rb *redisBackend) UpdateAssignments(ctx context.Context, req *pb.AssignTic
 		if err != nil {
 			return nil, errors.Wrap(err, "error sending ticket assignment set")
 		}
+
+		if callback != nil {
+			err = callback(ticket, ticket.Assignment)
```

Comment:
This isn't quite right - a ticket isn't actually assigned until after it is successfully set - if a ticket is deleted between retrieval and when it's set again with the proper assignment, then we correctly indicate that the ticket wasn't found, instead of reviving the ticket.

Reply:
Related to the comment above, if we return a list of tickets on UpdateAssignments we effectively have a list of the tickets that were successfully assigned.